tutorial/westeros/westeros_share_constraint.ipynb

```diff
@@ -0,0 +1,464 @@
+   "outputs": [],
+   "source": [
+    "# Example of index names of a set\n",
+    "scen.idx_names('map_shares_commodity_total')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Adding data to a set using index names\n",
+    "df = pd.DataFrame({'shares': [shares],\n",
+    "                   'node_share': country,\n",
+    "                   'node': country,\n",
+    "                   'type_tec': type_tec,\n",
+    "                   'mode': 'standard',\n",
+    "                   'commodity': 'electricity',\n",
+    "                   'level': 'secondary',\n",
+    "})\n",
+    "scen.add_set('map_shares_commodity_total', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 3: Defining technologies of share\n",
+    "Next, we define technologies that contribute towards fullfilling the share. We also need to set corresponding commodity and level for which the share constraint applies. In our example, the technology that provides 50% electricity from renewable energy is `wind_ppl`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "type_tec = 'electricity_renewable'\n",
+    "scen.add_cat('technology', type_tec, 'wind_ppl')\n",
+    "\n",
+    "# Reminder: use <scen.idx_names('map_shares_commodity_share')> to see the index names\n",
+    "df = pd.DataFrame({'shares': [shares],\n",
+    "                   'node_share': country,\n",
+    "                   'node': country,\n",
+    "                   'type_tec': type_tec,\n",
+    "                   'mode': 'standard',\n",
+    "                   'commodity': 'electricity',\n",
+    "                   'level': 'secondary',\n",
+    "})\n",
+    "scen.add_set('map_shares_commodity_share', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 4: Defining the share\n",
+    "Lastly, the actual share value needs to be added to the model. To represent a minimum share of 50% of total electricity generation from renewables, we use parameter `share_commodity_lo`. If this constraint should act as an upper bound, then parameter `share_commodity_up` can be used."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Fetching the index names of a parameter\n",
+    "scen.idx_names('share_commodity_lo')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Adding data to the parameter\n",
+    "df = pd.DataFrame({'shares': shares,\n",
+    "                   'node_share': country,\n",
+    "                   'year_act': [720],\n",
+    "                   'time': 'year',\n",
+    "                   'value': [.5],\n",
+    "                   'unit': '%'})\n",
```

Comment:
I suggest the unit of "-", as % can be misleading when the input values for the share are between 0 and 1.

Reply:
done